core/src/main/java/org/infinispan/stats/impl/ClusterCacheStatsImpl.java

```diff
@@ -87,22 +91,26 @@ public void injectDependencies(Cache<?, ?> cache, Configuration configuration) {
 
    @Start
    public void start() {
-      this.des = SecurityActions.getDefaultExecutorService(cache);
-   }
-
-   @Stop
-   public void stop() {
-      if (des != null && !des.isShutdown()) {
-         des.shutdownNow();
-      }
+      this.clusterExecutor = SecurityActions.getClusterExecutor(cache);
    }
 
    @Override
    void updateStats() throws Exception {
-      List<Map<String, Number>> responseList = new ArrayList<>();
-      List<CompletableFuture<Map<String, Number>>> responseFutures = des.submitEverywhere(new DistributedCacheStatsCallable());
-      for (CompletableFuture<Map<String, Number>> cf : responseFutures)
-         responseList.add(cf.get());
+      ConcurrentMap<Address, Map<String, Number>> resultMap = new ConcurrentHashMap<>();
+      TriConsumer<Address, Map<String, Number>, Throwable> triConsumer = (a, v, t) -> {
+         if (t != null) {
+            throw new CacheException(t);
```

Comment:
@danberindei do you think you could send me just a very simple template for a test class that puts the "cluster" of nodes in the states we need to watch out for. Then I can test against that with this and see what needs to be changed.

Reply:
Not sure if this is what you expected, but the first scenario would be:

1. get the cluster stats on node A and block the rpc
2. kill node B
2. unblock the rpc and check the results

I'm not very familiar with the stats, so all I can say about it is it shouldn't fail.

The second scenario would be:

1. stop the cache on node B (bonus scenario: add a blocking component so that the cache stays in the STOPPING state)
2. get the cluster stats on node A
3. check the results